### PR TITLE
Export open_mfdataset

### PR DESCRIPTION
**Pull request by meggart**
This documents and export the `open_mfdataset` function. Also, in conjunction with the recent YAXArrayBase PR this significantly decreases times for loading stacks of NetCDF files. Opening a stack of 50 NetCDFs went from 18 -> 0.8 seconds with this new version.

### CODE REVIEW

File: src/DatasetAPI/Datasets.jl [hunk 1]
```diff
@@ -337,6 +337,63 @@ open_mfdataset(g::AbstractString; kwargs...) = open_mfdataset(_glob(g); kwargs..
...
+open_mfdataset(DD.DimArray(files,DD.Ti()))
+````
+
+alternatively, if the dimension to concatenate along does not exist yet, the 
+dimension provided in the input arg is used:
+
+````julia
+files = ["a.nc","b.nc","c.nc"]
+open_mfdataset(DD.DimArray(files,DD.Dim{:NewDim}(["a","b","c"])))
+
```
Comment (lazarusA) [inline src/DatasetAPI/Datasets.jl:379]:
```suggestion
````
```
close the code-block.

**Review by lazarusA — CHANGES_REQUESTED** [review]
just close the code-block, and CI for docs should be fine.

File: src/DatasetAPI/Datasets.jl [hunk 2]
```diff
@@ -337,6 +337,63 @@ open_mfdataset(g::AbstractString; kwargs...) = open_mfdataset(_glob(g); kwargs..
...
+    newshape = ntuple(i->i!=istack ? 1 : length(alldiskarrays),ndims(firstcube))
+    newda = DiskArrays.ConcatDiskArray(reshape(alldiskarrays,newshape))
+    newdims = Base.setindex(firstcube.axes,newdim,istack)
+    YAXArray(newdims,newda,deepcopy(firstcube.properties))
+end
+
+"""
+    open_mfdataset(files::DD.DimVector{<:AbstractString}; kwargs...)
+
+Opens an concatenates a list of dataset paths along the dimension specified in `files`.
```
[the remark this answers is not in the document]
Comment (Balinus) [reply src/DatasetAPI/Datasets.jl:362]: Can an "external" dimension be specified? For example, say we have 100 files with 2D matrix and we would like to add a time dimension and `cat` the matrices based on that? (in other words, the dimension is not present in `files`).

File: src/DatasetAPI/Datasets.jl [hunk 3]
```diff
@@ -337,6 +337,63 @@ open_mfdataset(g::AbstractString; kwargs...) = open_mfdataset(_glob(g); kwargs..
...
+````
+
+alternatively, if the dimension to concatenate along does not exist yet, the 
+dimension provided in the input arg is used:
+
+````julia
+files = ["a.nc","b.nc","c.nc"]
+open_mfdataset(DD.DimArray(files,DD.Dim{:NewDim}(["a","b","c"])))
+
+"""
```
Comment (lazarusA) [inline src/DatasetAPI/Datasets.jl:380]:
```suggestion
````julia
files = ["a.nc","b.nc","c.nc"]
open_mfdataset(DD.DimArray(files,DD.Dim{:NewDim}(["a","b","c"])))
````
"""
```

**Review by lazarusA — APPROVED** [review]

**Pull request merged**